packages/client-generator-ts/tests/assert-typescript-is-valid.ts

```diff
@@ -0,0 +1,203 @@
+import { match } from 'ts-pattern'
+import ts from 'typescript'
+import { expect } from 'vitest'
+
+interface AssertTypeScriptIsValidInput {
+  files: {
+    [filename: string]: string
+  }
+  moduleFormat: 'cjs' | 'esm'
+}
+
+/**
+ * Asserts that the provided TypeScript code is syntactically and semantically valid.
+ */
+export function assertTypeScriptIsValid({ files, moduleFormat }: AssertTypeScriptIsValidInput) {
+  const configFromModuleFormat = match(moduleFormat)
+    .with('cjs', () => ({
+      module: ts.ModuleKind.Node16,
+      moduleResolution: ts.ModuleResolutionKind.Node16,
+      types: ['node'],
+    }))
+    .with('esm', () => ({
+      module: ts.ModuleKind.ES2022,
+      moduleResolution: ts.ModuleResolutionKind.Bundler,
```

Comment:
Should we also test Node.js + ESM? (with both `module` and `moduleResolution` set to either `Node16` or `NodeNext`)